lib/api.js

```diff
@@ -104,7 +107,103 @@ module.exports = {
    * }
    * ```
    */
-  messages: []
+  // BUG previous messages aren't cleaned up
+  // messages: [],
+  /**
+   * Messages Error API
+   *
+   * push plugin errors to result.messages for reporting
+   *
+   * @memberof tree
+   *
+   * @method error
+   *
+   * @param  {String} plugin   Name ('posthtml-plugin')
+   * @param  {String} message  Error
+   *
+   * @return {Object} PostHTML Error Message
+   *
+   * ```js
+   * tree.error('posthtml-plugin', 'Error', node.location)
+   * ```
+   */
+  error: function (plugin, message, location) {
+    const err = new PluginError({
+      src: this.src,
+      file: this.options.from,
```

Comment:
`file` => `from`

Reply:
if this not file ?

Reply:
It's the 'parent' or the file processed (the source), which is most of the time === `options.from` (if set). I used `file` in `tree.dependency` as the prop for to the path to the `dependency` and `from` for the parent, which should be equally handled in `warnings` && `errors` for API consistency.

Reply:
```js
var myCustomString = `<html>...</html>`;

posthtml().process(myCustomString)
```